pylabnet/launchers/launcher.py

```diff
@@ -107,12 +106,25 @@ def __init__(self, script=None, server_req=None, gui_req=None, auto_connect=True
             self.log_ip = self.args['logip']
             self.log_port = int(self.args['logport'])
             self.num_clients = int(self.args['numclients'])
+            self.debug = int(self.args['debug'])
+            self.server_debug = int(self.args['server_debug'])
+            self.gui_debug = int(self.args['gui_debug'])
         except IndexError:
             raise
 
-        # Connect to logger
+        # Connect to logger.
         self.logger = self._connect_to_logger()
 
+        # Halt execution and wait for debugger connection if debug flag is up.
+        if self.debug == 1:
+            import ptvsd
+            import os
+            # 5678 is the default attach port in the VS Code debug configurations
+            self.logger.info(f"Waiting for debugger to attach to PID {os.getpid()} (launcher)")
```

Comment:
is there ever a scenario where we would want to de remote debugging?

Reply:
There might be. I do think this should be doable using this code and a modified debugger configurations:
![image](https://user-images.githubusercontent.com/4958226/82462339-7279bf80-9a89-11ea-9f10-08530b18e18d.png)

From [Python debugging in VS code](https://code.visualstudio.com/docs/python/debugging).